### PR TITLE
feat: supporting react 18

### PR DESCRIPTION
**Pull request by erunion**
## 🧰 Changes

* [x] Allowing installs on React 18 codebases.

## 🧬 QA & Testing

All React 18 tests should be passing!

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -7,33 +7,35 @@ jobs:
...
       - name: Install deps
         run: npm ci
 
-      - name: Install React 16
-        if: matrix.react == '16'
+      - name: Install React ${{ matrix.react }}
         run: npm i react@${{ matrix.react }} react-dom@${{ matrix.react }}
 
+      # `@testing-library/react@13` dropped support for React 16 and 17 so to ensure that we run
+      # fine under React 18 we should upgrade it if we're testing React 18.
```
Comment (kellyjosephprice) [inline .github/workflows/ci.yml:35]: That's annoying, good note though.

Comment (kellyjosephprice) [reply .github/workflows/ci.yml:35]: Would this be better handled by a `postinstall` script? That way it works locally too?

Got the idea here: https://stackoverflow.com/a/60603000/659661

**Review by kellyjosephprice — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (erunion) [reply .github/workflows/ci.yml:35]: I'm weary of putting it in a postinstall script because those run when you install the package from npm.

Comment (kellyjosephprice) [reply .github/workflows/ci.yml:35]: Oh, TIL. I should probably learn more about npm...

**Pull request merged**